components/compiler-families/intel-compilers-devel/SOURCES/oneapi-modgen.lua

```diff
@@ -0,0 +1,937 @@
+#!/usr/bin/env lua
+--[[
+===============================================================================
+This script is part of the OpenHPC project.
+
+It may have been modified from the default version supplied by the underlying
+release package (if available) in order to apply patches, perform customized
+build/install configurations, and supply additional files to support
+desired integration conventions.
+
+Licensed under the Apache License, Version 2.0 (the "License"); you
+may not use this file except in compliance with the License. You may
+obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+implied. See the License for the specific language governing
+permissions and limitations under the License
+===============================================================================
+]]
+
+local function description()
+    return [[
+ONEAPI-MODGEN
+    OpenHPC - Lmod Module Generator for Intel(R) oneAPI Toolkits
+    
+    Creates and manages OpenHPC Lmod module creation for Intel(R) oneAPI
+    toolkits and legacy software tools. The default installation path
+    /opt/intel/oneapi is used to locate installed versions. Other
+    installations can be added manually using the root path for that
+    installation.
+    ]]
+end
+
+require "io"
+require "os"
+require "lfs" -- lua-filesystem RPM required
+
```

Comment:
I do not see any code directly interacting with the lmod lua code base. This just seems to use three lua modules which are not connected to lmod. I see it problematic introducing a new language into the OpenHPC code base.

The modules `os`, `io` and `lfs` are probably also implementable in something which fits the current repository better.

Reply:
Are you offering to convert and maintain in a different language?
One other option I can support is to use C and change this to a binary executable.